### ISSUE TITLE
Lint-fixing examples can break them

### ISSUE THREAD

**Issue by edg2s** [wikimedia/eslint-docgen#33]
In plugin-mediawiki we have two tests:
```js
new OO.ui.ButtonWidget( { classes: ["foo"] } )
new OO.ui.ButtonWidget( { "classes": ["foo"] } )
```
The quoted key case gets fixed to be identical to the one above.

In this particular case it can just be noDoc'd but in other cases this may not be appropriate.

Lint-fixing can be turned off globally but that too may not be desirable.

Possible solutions:
* Add a per rule noDocFix option
* Restrict fixing to just whitespace fixes
* Have a mode where eslint-docgen reports lint errors in examples, but doesn't fix them. Properly this would be a pre-processing plugin.

**Comment by edg2s** [comment]
Added "Warn when duplicate tests are detected". This warning could suggest possible solutions.
